assembly/assembly-main/src/assembly/bin/launcher.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/sh
+
+# See: https://sipb.mit.edu/doc/safe-shell/
+set -e
+#set +o posix
+
+# Run the finish function if exit signal initiated
+trap exit SIGHUP SIGINT SIGTERM
+
+init_global_variables () {
+
+  # For coloring console output
+  BLUE='\033[1;34m'
+  GREEN='\033[0;32m'
+  NC='\033[0m'
+
+  ### Define various error and usage messages
+  USAGE="
+Usage:
+  launcher [COMMAND]
+     start                              Starts Che server in new console
+     stop                               Stops Che server
+     restart                            Restart Che server
+     update                             Pull latest version of codenvy/che
+
+Docs: http://eclipse.org/che/getting-started.
+"
+
+  DOCKER_HOST_IP=""
+  DEBUG=false
+  CHE_LIB_VOLUME_PATH="/home/user/che/lib"
+  CHE_WS_VOLUME_PATH="/home/user/che/workspaces"
+  CHE_STG_VOLUME_PATH="/home/user/che/storage"
+  DOCKER_SOCKET="/var/run/docker.sock"
+  CHE_SERVER_PORT="8080"
+  CHE_IMAGE_VERSION="nightly"
+}
+
+usage () {
+  echo "${USAGE}"
+}
+
+error_exit () {
+  echo
+  echo "!!!"
+  echo -e "!!! ${1}"
+  echo "!!!"
+  echo
+}
+
+get_docker_host_ip() {
+  DOCKER_HOST_IP=$(docker run --rm --net host \
+                    alpine sh -c \
+                    "ip a show eth0" | \
+                    grep 'inet ' | \
+                    cut -d/ -f1 | \
+                    awk '{ print $2}')
+}
+
+parse_command_line () {
+
+  if [ $# -eq 0 ]; then
+    usage
+    return
+  fi
+
+  for command_line_option in "$@"
+  do
+  case ${command_line_option} in
```

Comment:
Personally I prefer leaving `do` in the same line where `for` is placed and addition of indentation of body of cycle.
It makes shell script more similar to java code. Please consider such reformatting.